### DISCUSSION TITLE
Trying to pull from git with merge conflicts from previously unmerged PR

### DISCUSSION THREAD

**Discussion by rperry99** [EddieHubCommunity/support#184] [Q&A] [answered]
So I am trying to add my name and projects to this list here: [Zero to Mastery JavaScript20 Projects](https://github.com/zero-to-mastery/javascript20-projects). 
Here is [my forked repo](https://github.com/rperry99/javascript20-projects/tree/russ-updates)

I had made a PR to Zero to Mastery's repo about 20 days ago and it has not been merged / checked / or anything. The first PR is PR number 98.

So I made the first PR 20 days ago, which added about 4 or 5 of my projects to the readme as well as add live links to the projects I already added. As I said, this was not merged or reviewed.

I did a Git Pull from the master branch and here is where there are merge conflicts. I see when I did the git pull, it is pulling the versions of my links without the live links. 

![image](https://user-images.githubusercontent.com/24259120/93138529-25ab9b80-f6ad-11ea-856f-5be4b499f647.png)

So in the above example, (I am Russ Perry), The old version that I changed with PR1 is in green. The pull request I am trying to do is what is in blue, which has my old version along with someone else who had thier PR merged before mine. 

I am not sure how to handle this. If I accept current change, it gets rid of the other person's commit. If I accept incoming, I lose everything I did before. I tried accepting both but I think I broke something when I did that.

I have never done a merge conflict before, so I am not really sure how to handle this.

Looking at my old PR, it looks like it automatically created a second PR with changes and saying there were conflicts? I didn't push anything to this repo after I tried merging.

**Comment by eddiejaoude** [comment]
You can accept both, by manually merging them, I will create pseudo example:

## Conflict 

```diff
<<<<<<< HEAD
+ Russ Perry - [GitHub](GitHub.com) [ViewLive[(netlify.com)
=======
+ Josue Cedenoo - [GitHub](GitHub.com) [ViewLive[(netlify.com)
+ Russ Perry - [GitHub](GitHub.com)
>>>>>>> abcdef
```

## Resolution 

Keep their changes and the changes you want

```diff
+ Josue Cedenoo - [GitHub](GitHub.com) [ViewLive[(netlify.com)
+ Russ Perry - [GitHub](GitHub.com) [ViewLive[(netlify.com)
```

Once you are happy with the manual merge conflict, you can do 

1. `git add <files>` 
2. `git commit -I -m "--- MESSSAGE ---" <files>`

---

## Side note

Does your new PR rely on the previous PR you created?

### If `no`

- I would create a new branch from their default branch and do work as if the previous PR wasn't created

### If `yes`

- Merge changes manually
- If multiple commits and no one else working in your branch, you could `rebase`

**Comment by rperry99** [comment, marked as answer]
@eddiejaoude  So my new changes do **not** rely on my first one being accepted. Do I need to re-fork their project? Or do I just create a new branch entirely without pulling what they have? If I pull what they have, I get the merge conflict.

**Reply by eddiejaoude** [reply]
I think I know why you get the conflict, are you working on your fork in their default branch `master`?

The default branch should not have any of your changes otherwise it will get conflicts, especially if they squash commits when merging PRs.

I recommend:
1. create a new branch from your `master` branch in your fork - this should contain your previous changes 
2. close your PR (if possible, for example there is no chat going on), create a new PR with your new branch from `step 1`
3. delete your local `master` branch and fetch it again - this will make sure your `master` is the same as their `master` (`fork.master` === `upstream.master`), then your they can always be in sync without conflict
4. create a new branch from new `master` for your new changes - this will ensure both PRs are independent, otherwise your 2nd PR will always contain the 1st PR's changes and be blocked by the 1st PR

I hope that makes sense? If not let me know 🤓

**Reply by rperry99** [reply]
Should my master be the same as the Zero to Mastery master? Becuase its not. My master right now is actually really empty. I was never able to get my master to mach theirs. Bash yells at me when I try to pull.

**Reply by eddiejaoude** [reply]
Yes default branches should match between the fork and upstream (source) repos - your `master` is probably just out of date. 

It would have matched theirs when you first forked the repo. Your latest commit in `master` (excluding your changes), should be in their history on `master` too - you might have to look further back to find it though.

I would still recommend follow the `4 steps` above, and you will still need to manually merge the changes in your new branch too as I mentioned here https://github.com/EddieJaoudeCommunity/support/discussions/184#discussioncomment-78910

**Reply by eddiejaoude** [reply]
btw I did a video a few months ago on how to resolve merge conflicts, it might help, any questions let me know https://youtu.be/W--RpZcQ7FA

**Reply by rperry99** [reply]
Thank you! That worked! I was able to push with no issues!

I followed your 4 steps listed above and it worked perfectly!

**Reply by eddiejaoude** [reply]
Great 👍 - I might do a video on this because I get asked this a lot. I think it is a problem most people have - thank you 🤓
